dbeam-core/src/main/scala/com/spotify/dbeam/JdbcAvroRecord.java

```diff
@@ -62,13 +63,14 @@ public static GenericRecord convertResultSetIntoAvroRecord(
 
   public static Map<Integer, SQLFunction<ResultSet, Object>> computeAllMappings(ResultSet resultSet)
       throws SQLException {
-    ResultSetMetaData meta = resultSet.getMetaData();
-    Map<Integer, SQLFunction<ResultSet, Object>> mappings = new HashMap<>();
+    final ResultSetMetaData meta = resultSet.getMetaData();
+    final int columnCount = meta.getColumnCount();
+    final Map<Integer, SQLFunction<ResultSet, Object>> mappings = new HashMap<>(columnCount);
 
-    for (int i = 1; i <= meta.getColumnCount(); i++) {
+    for (int i = 1; i <= columnCount; i++) {
       mappings.put(i, computeMapping(meta, i));
     }
-    return Collections.unmodifiableMap(mappings);
+    return ImmutableMap.copyOf(mappings);
```

Comment:
`computeMapping` can return null (which we should try to avoid)
`ImmutableMap` doesn't allow null values, `Collections.unmodifiableMap` allows null values. 
This would hence break with a `NullPointerException`

Shall we add test cases for these?

Reply:
> computeMapping can return null (which we should try to avoid)

It does not. It returns Functions that might return null. Or am I missing something?

Reply:
Ahh, my bad. I had missed out this.